lib/ohai/plugins/sysconf.rb

```diff
@@ -0,0 +1,43 @@
+#
+# Author:: Davide Cavalca <dcavalca@fb.com>
+# Copyright:: Copyright (c) 2016 Facebook
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+Ohai.plugin(:Sysconf) do
+  provides "sysconf"
+
+  collect_data do
+    getconf_path = which("getconf")
+    if getconf_path
+      getconf = shell_out("#{getconf_path} -a")
+
+      sysconf Mash.new unless sysconf
```

Comment:
wrap from here down in `if getconf.exitstatus == 0` so we don't create `sysconf` or bother parsing it if we have a getconf that does't do what we expect, like darwin lacks a `-a` flag.